hammer/sim/xcelium/__init__.py

```diff
@@ -377,6 +381,336 @@ def generate_sim_tcl(self) -> bool:
+
+    # Process non-specified standard AMS opts
+    for (opt, setting) in ams_opts_proc.items():
+      if (opt not in bool_list and setting is not None):
+        ams_opts_proc [opt] = f"-{opt} {setting}"
+      
+    # Convert AMS opt dictionary into formatted string
+    ams_opt_str = ""
+    for (opt, setting) in ams_opts_proc.items():
+      ams_opt_str += f"{setting}\n"
+
+    ### Read in disciplines file, if it exists.
+    disciplines = self.get_setting("sim.xcelium.disciplines")
+
+    if (disciplines):
+      df = open(disciplines, "r")
+      discpline_str = df.read() + "\n"
+      df.close()
+    else:
+      discpline_str = ""
+
+    ### Read in A/MS schematics, control files, connect libs, and convert them into formatted strings
+    cwd = os.getcwd()
+    schematicdir = os.path.join(cwd, self.get_setting("sim.xcelium.schematics"))
+    connectdir = os.path.join(cwd, self.get_setting("sim.xcelium.connectlibs"))
+    #acf = os.path.join(cwd, self.get_setting("sim.xcelium.acf")) + "\n"
+    amscf = os.path.join(cwd, self.get_setting("sim.xcelium.amscf")) + "\n"
+
+    # Retrieve formatted string of schematic files
+    schematics = self.retrieve_files(schematicdir, exts=[".sp", ".vams"])
+
+    # Retrieve formatted string of connect files
+    connectlibs = self.retrieve_files(connectdir, exts=[".vams"])
+
+    ### Attach additional options 
+    additional_opts = "-spectre_args \"++aps\"\n-messages \n-ieinfo \n"
+
+    ### Create and return combined string 
+    ams_opt_header = "# AMS OPTS:\n"
+    formatted_opts = ams_opt_header + ams_opt_str + discpline_str + additional_opts + "\n"
+
+    ams_file_header = "# AMS FILES:\n"
+    formatted_files = ams_file_header + amscf + connectlibs + "\n"
+    
+    return formatted_files + formatted_opts
+
+  def attach_opts(self, filepath, attachment):
+    f = open(filepath, "a+")
+    f.write(attachment)
+    f.close
+    return 
+  
+  def get_disciplines(self) -> str:
+    ### Read in disciplines file, if it exists.
+    disciplines = self.get_setting("sim.xcelium.disciplines")
+    cwd = os.getcwd()
+    dpath = os.path.join(cwd, disciplines)
+    if disciplines:
+      df = open(dpath, "r")
+      discipline_opts = df.read() + "\n"
+      df.close()
+      return discipline_opts
+    else:
+      return ""
+
+    
+
+    if (disciplines):
+      dpath = os.path.join(cwd, disciplines)
+      df = open(dpath, "r")
+      discpline_str = df.read()
+      disciplines = "\n".split(discpline_str)
+      discipline_opts = [opt for opt in disciplines if opt != ""]
+      df.close()
+    else:
+      discipline_opts = []
+
+    if discipline_opts:
+      discipline_opts_proc = [f"-SETD \"{opt}\"" for opt in discipline_opts]
+    else:
+      discipline_opts = []
+
+    return discipline_opts_proc
+  
+  def extract_ams_opts(self, step):
+    ams_opts = self.get_setting(f"{self.tool_config_prefix()}.ams_opts", [])
+    cwd = os.getcwd()
+    match (step):
```

Comment:
Technically, match/case is Python 3.10+. We can discuss if we want to bump the minimum Python version.

Reply:
This is actually an outdated method from when I was trying to have AMS to follow the same three-step method as the base plugin. It'll probably be removed in the upcoming commit. As for the minimum python version more generally, I don't know if any of the features of 3.10 are worth forcing people with 3.9 to update the version.